specs/Checkbox.md

```diff
@@ -0,0 +1,388 @@
+# Checkbox component specification
+
+The `Checkbox` component allows a user to choose between two mutually exclusive options. 
+
+## Related variant considerations
+
+- Toggle: should be a separate component because it has different anatomy & warrants unique themability.
+
+- Indeterminate/Tri State: should support groups with hierarchical checkboxes where the parent checkbox can be in a mixed state if its children checkboxes aren't all checked (and if they are all checked, then the parent is checked; same if they are all unchecked, then the parent is unchecked).
+
+## Reference implementations
+
+https://codesandbox.io/s/checkboxes-ggpx1
+
+Note about the Stardust example: there's some weirdness with how the theme providers are interacting with each other, the Stardust checkbox's styling is messing up as a result.
+
+Fabric Checkbox [docs](https://developer.microsoft.com/en-us/fabric#/controls/web/Checkbox)
+
+Stardust Checkbox [docs](https://microsoft.github.io/fluent-ui-react/components/checkbox/definition)
+
+Material UI Checkbox [docs](https://material-ui.com/components/checkboxes/)
+
+BaseUI Checkbox [docs](https://baseweb.design/components/Checkbox/)
+
+Chakra Checkbox [docs](https://chakra-ui.com/Checkbox)
+
+Cabon Checkbox [docs](https://www.carbondesignsystem.com/components/checkbox/code)
+
+AntD Checkbox [docs](https://ant.design/components/Checkbox/)
+
+FastDNA Checkbox [docs](https://explore.fast.design/components/Checkbox)
+
+
+## Props
+
+> TODO: Consult the prop wizard to derive consistently defined props.
+
+| Name | Type | Default value |
+| ---- | ---- | ------------- |
+
+
+### Recommended props
+
+| Name                 | Type                                                        |
+| --------------       | ----------------------------------------------------------- |
+| ariaDescribedBy      | string                                                      |
+| ariaLabel            | string                                                      |                                                
+| ariaLabelledBy       | string                                                      |                                                
+| as                   | React.ElementType                                           |
```

Comment:
All `as` props should probably be `keyof JSX.IntrinsicElements` in order to limit it to native HTML elements and exclude React components.